group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2015,14 +2024,6 @@ public void replay(Record record, GroupType groupType) {
                 );
                 break;
 
-            case 5:
```

Comment:
I'm surprised by this removal. Doesn't it break replay for consumer groups?

Reply:
At least, perhaps this method need a comment since I think it's used to drive replay for records which can be used for different group types.

Reply:
So we moved from ConsumerMemberMetadata (5) to ShareMemberMetadata (10) so we do not need to handle case of 5 for share group hence I have moved the handling to GroupCoordinatorShard. The method in GroupMetadataManager handles common cases for consumer and share group.

Reply:
Yes, I see.